WelsonJS.Toolkit/WelsonJS.Launcher/ResourceServer.cs

```diff
@@ -164,26 +164,27 @@ private async Task<bool> ServeBlob(HttpListenerContext context, string path, str
 
                 try
                 {
-                    using (var client = new HttpClient())
+                    HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, url);
+                    var ua = context?.Request?.UserAgent;
+                    if (!string.IsNullOrEmpty(ua))
                     {
-                        HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, url);
-                        request.Headers.UserAgent.ParseAdd(context.Request.UserAgent);
-                        HttpResponseMessage response = await client.SendAsync(request);
+                        request.Headers.UserAgent.ParseAdd(ua);
+                    }
+                    HttpResponseMessage response = await _httpClient.SendAsync(request);
 
-                        if (!response.IsSuccessStatusCode)
-                        {
-                            _logger?.Error($"Failed to serve blob. URL: {url}, Status: {response.StatusCode}");
-                            return false;
-                        }
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        _logger?.Error($"Failed to serve blob. URL: {url}, Status: {response.StatusCode}");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Avoid string interpolation within log messages, as it can blur the lines between variables and the log content. Instead, opt for structured logging, where variables are passed as separate arguments. This improves the clarity of logs and facilitates efficient log analysis. For more information, refer to https://github.com/NLog/NLog/wiki/How-to-use-structured-logging.